tests/playwright-test/expect.spec.ts

```diff
@@ -1000,3 +1000,42 @@ test('should respect timeout from configured expect when used outside of the tes
   expect(stdout).toBe('');
   expect(stripAnsi(stderr)).toContain('Timed out 10ms waiting for expect(locator).toBeAttached()');
 });
+
+test('should expose timeout to custom matchers', async ({ runInlineTest, runTSC }) => {
+  const files = {
+    'playwright.config.ts': `
+      export default {
+        expect: { timeout: 1100 }
+      };
+    `,
+    'a.test.ts': `
+      import type { ExpectMatcherState, MatcherReturnType } from '@playwright/test';
+      import { test, expect as base } from '@playwright/test';
+
+      const expect = base.extend<{assertTimeout: (this: ExpectMatcherState, page: any, value: number) => MatcherReturnType}>({
```

Comment:
Typescript should infer things:

```suggestion
      const expect = base.extend({
```